### ISSUE TITLE
How to put Apps in specific namespaces?

### ISSUE THREAD

**Issue by anjuls** [kalmhq/kalm#139]
The issue I am seeing is that `kalm` is very opinionated. It doesn't give me option to provide namespace. For each app, a new namespace is created. 

Also, How can I browse existing applications, deployments on my cluster through kalm?

**Comment by tstyle11** [comment]
Yes Kalm indeed makes a rather opinionated decision in that each app gets a distinct namespace. When we made this decision, we realized it will probably turn a subset of users off. But we've found enough upside in practice (etc.. getting apps up and running quicker, reducing boilerplate, keeping the UI simple) that we want to try going down this route a bit longer.

In an earlier version of Kalm, we had apps and namespaces as separate concepts, and you can have a namespace containing multiple apps. But most people who used Kalm simply created 1 namespace to hold 1 app, or was confused as just put everything in the default namespace. The extra layer of abstraction simply caused more confusion than it was worth.

Are there specific ways you'd like to use namespaces that gets blocked by this app=namespace convention? Really appreciate the feedback!

**Comment by infrar** [comment]
Yes Kalm indeed makes a rather opinionated decision in that each app gets a distinct namespace. When we made this decision, we realized it will probably turn a subset of users off. But we've found enough upside in practice (etc.. getting apps up and running quicker, reducing boilerplate, keeping the UI simple) that we want to try going down this route a bit longer.

In an earlier version of Kalm, we had apps and namespaces as separate concepts, and you can have a namespace containing multiple apps. But most people who used Kalm simply created 1 namespace to hold 1 app, or was confused as just put everything in the default namespace. The extra layer of abstraction simply caused more confusion than it was worth.

Are there specific ways you'd like to use namespaces that gets blocked by this app=namespace convention? Really appreciate the feedback!

**Comment by infrar** [comment]
Regarding browsing apps: all apps created via Kalm should be visible in the _/applications_ page. Deployments created outside of Kalm are currently not visible, but will be in a later version.

**Comment by anjuls** [comment]
I was thinking that I can use kalm on top of my existing cluster and apps. I feel it is a very basic Kubernetes building block and should be available for selection. From the first impression, kalm seems good for those who don't want to touch their YAMLs.

**Comment by davidqhr** [comment]
#138

**Comment by evanrich** [comment]
@infrar  one possible use case, is to group seperate apps together.  For example, I have a "Media" namespace, that contains sonarr, radarr, jackett.  I then have a network policy that then only allows traffic in to these pods.    For simplicity sake, it's easier to group those 3 pods into 1 namespace, then to have "sonarr" "radarr" "jackett" namespace...etc
